client/components/SearchBar.js

```diff
@@ -0,0 +1,94 @@
+import React, { useState, useEffect } from 'react';
+import { fetchProducts } from '../store/products';
+import { connect } from 'react-redux';
+import { Link } from 'react-router-dom';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faMagnifyingGlass } from '@fortawesome/free-solid-svg-icons';
+import Button from 'react-bootstrap/Button';
+
+function SearchBar(props, { placeholder }) {
+  let [filteredData, setFilteredData] = useState([]);
+  let [wordEntered, setWordEntered] = useState('');
+  let [products, setProducts] = useState(props.allProducts);
+
+  useEffect(() => {
+    props.getProducts();
+    setProducts(props.products);
```

Comment:
We should never do something like this. when getProducts is dispatched, props.products is still an empty array, so we're always just going to setProducts to an empty array here, making the line redundant